public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -158,8 +170,40 @@ export class CloudWatchDatasource extends DataSourceWithBackend<CloudWatchQuery,
       region: this.replace(this.getActualRegion(target.region), options.scopedVars, true, 'region'),
     }));
 
+    const retryStrategy = ({
+      maxRetryAttempts = 3,
+      scalingDuration = 1000,
+    }: {
+      maxRetryAttempts?: number;
+      scalingDuration?: number;
+    } = {}) => (attempts: Observable<any>) => {
+      return attempts.pipe(
+        mergeMap((error, i) => {
+          const isLimitError = typeof error === 'string' && error.includes('LimitExceededException');
```

Comment:
I check the error string here, not sure if we want to go into more explicit error codes etc or some translation between client codes to some generic Grafana codes etc.